Exercise08/Program.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Exercise08
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            var numbers1 = inputNumbers(new[] { 1, 2, 3, 4, 5, 6 }).ToList();
+            var numbers2 = inputNumbers(new[] { 4, 5, 6, 7, 8, 9 }).ToList();
+
+            var listCommon = numbers1.Join(
+                numbers2,
+                x => x,
+                y => y,
+                (x, y) => new
+                    {
+                        x = x
+                    })
+                .ToList();
+
+            listCommon.ForEach(Console.WriteLine);
+            Console.WriteLine();
+
+            var listOfNumberOnlyExistListFirst = numbers1
+                .Where(x => !numbers2
+                    .Any(y => y == x))
+                .ToList();
+
+            listOfNumberOnlyExistListFirst.ForEach(Console.WriteLine);
+            Console.WriteLine();
+
+            var listOfNumberExistOnlyList = numbers1
+                .Union(numbers2)
+                .Where(x => !listCommon
+                    .Any(y => y.x == x))
+                .ToList();
+
+            listOfNumberExistOnlyList.ForEach(Console.WriteLine);
+            Console.WriteLine();
+
+
+            var sum = numbers1.Select((x, i) => new
+                {
+                    x = x,
+                    i = i
+                })
+                .TakeWhile(x => true)

```

Comment:
`.TakeWhile(x => true)` and ` .TakeWhile(y => true)` is redundance. Try to simple it
Here is some suggestions
```C#
var total2 = Enumerable.Range(0, numbers1.Count * numbers2.Count)
    .Where(x => x / numbers1.Count != x % numbers2.Count)
    .Sum(x => numbers1[x / numbers1.Count] * numbers2[x % numbers2.Count]);

var total3 = numbers1.Sum() * numbers2.Sum() -
            Enumerable.Range(0, Math.Min(numbers1.Count, numbers2.Count))
                .Sum(x => numbers1[x] * numbers2[x]);

var total4 = Enumerable.Range(0, numbers1.Count)
    .Sum(x => Enumerable.Range(0, numbers2.Count).Where(y => x != y).Sum(y=>numbers1[x] * numbers2[y]));
```

Reply:
Done